packages/grafana-ui/src/components/DateTimePickers/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { FC, FormEvent, ReactNode, useCallback, useEffect, useState } from 'react';
+import { useMedia } from 'react-use';
+import Calendar from 'react-calendar/dist/entry.nostyle';
+import { css } from '@emotion/css';
+import { dateTimeFormat, DateTime, dateTime, GrafanaTheme2 } from '@grafana/data';
+import { Button, Field, Icon, Input, Portal } from '../..';
+import { ClickOutsideWrapper } from '../../ClickOutsideWrapper/ClickOutsideWrapper';
+import { isValid } from '../utils';
+import { getBodyStyles, getStyles as getCalendarStyles } from '../TimeRangePicker/TimePickerCalendar';
+import { useStyles2, useTheme2 } from '../../../themes';
+import { TimeOfDayPicker } from '../TimeOfDayPicker';
+
+export interface Props {
+  /** Input date for the component */
+  date: DateTime;
```

Comment:
This will make the date required, but what if we have a use case where we want it to be undefined? An example is a form here datetime field can be empty.